lib/SIL/IR/SILDeclRef.cpp

```diff
@@ -981,6 +981,10 @@ bool SILDeclRef::isForeignToNativeThunk() const {
   // have a foreign-to-native thunk.
   if (!hasDecl())
     return false;
+  // A default argument generator for a C++ function is a Swift function, so no
+  // thunk needed.
+  if (isDefaultArgGenerator())
+    return false;
```

Comment:
So why is it marked as foreign?

Reply:
The parent function is marked as foreign because it has a Clang node, but the default argument generator isn't foreign, and doesn't require a thunk despite its parent function being foreign. This check needs to happen before the `requiresForeignToNativeThunk` check below causes us to assume that a thunk is needed.

Reply:
Hello How are you @egorzhdan @rjmccall @hyp @Oudinunit @lilyball @drodriguez @drodriguez @mattt @xwu @jtbandes @erica